cmd/anonymizer/app/writer/writer_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package writer
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+var tags = []model.KeyValue{
+	model.Bool("error", true),
+	model.String("http.method", "POST"),
+	model.Bool("foobar", true),
+}
+
+var traceID = model.NewTraceID(1, 2)
+
+var span = &model.Span{
+	TraceID: traceID,
+	SpanID:  model.NewSpanID(1),
+	Process: &model.Process{
+		ServiceName: "serviceName",
+		Tags:        tags,
+	},
+	OperationName: "operationName",
+	Tags:          tags,
+	Logs: []model.Log{
+		{
+			Timestamp: time.Now(),
+			Fields: []model.KeyValue{
+				model.String("logKey", "logValue"),
+			},
+		},
+	},
+	Duration:  time.Second * 5,
+	StartTime: time.Unix(300, 0),
+}
+
+func TestNew(t *testing.T) {
+	nopLogger := zap.NewNop()
+	tempDir := t.TempDir()
+
+	tests := []struct {
+		name   string
+		config Config
+	}{
+		{
+			name: "successfully create writer",
+			config: Config{
+				MaxSpansCount:  10,
+				CapturedFile:   tempDir + "/captured.json",
+				AnonymizedFile: tempDir + "/anonymized.json",
+				MappingFile:    tempDir + "/mapping.json",
+			},
+		},
+		{
+			name: "captured.json doesn't exist",
+			config: Config{
+				CapturedFile:   tempDir + "/nonexistent_directory/captured.json",
```

Comment:
I don't understand - how is this different from the above case? There are no files in `tempDir` in each test.

Reply:
This was to increase the code coverage in the tests. The nonexistent directory doesn't exist inside the `tempDir` so it would force an error and increase the code coverage.